### DISCUSSION TITLE
Add token is source

### DISCUSSION THREAD

**Discussion by sylvain258** [maplibre/maplibre-gl-js#4902] [Q&A]
Hello,

I would like to know how to allow a source to access a VT url with an authorization token. I have an API that produces VTs but only allows access if you have a token.
I found in map options, there is an option "transformRequest" where it's possible to put an access token in the header. But it doesn't work.
`            transformRequest: (url, resourceType) => {
                if (resourceType === 'Source' && url.startsWith('http://myHost')) {
                return {
                    url: url,
                    headers: { 'Authorization': 'Bearer ' + yourAuthToken }
                }
                }
            }`
Thanks

**Comment by HarelM** [comment]
It is not clear what "it doesn't work" mean, tarnsfromRequest is the means to achieve the required behavior.

**Comment by sylvain258** [comment]
I inserted this "transformRequest" in my code, in the map declaration. I can't display my VT layer. I have a return that my token is not found.

**Reply by HarelM** [reply]
Did you check the network tab to see what is being sent?

**Reply by sylvain258** [reply]
The problem seems to come from to several errors about VT rendering. I have many errors like "Unimplemented type" with values 3, 4, 6 and 7. I don't know that is mean. My VTs are generated correctly, it seems it's an error in the rendering.

**Reply by HarelM** [reply]
I'll be surprised if this is a rendering issue as we have many examples working correctly, please check your data thoroughly...

**Reply by sylvain258** [reply]
It's curious because that works correctly until now, few days ago it is not possible to display my layers. And for an unknown reason, I have a layer displays again. And I modified nothing.
